tests/Console/E2ETest.php

```
@@ -0,0 +1,243 @@
+    }
+
+    /**
+     * @group e2e
+     */
+    public function test_it_runs_configure_command_if_no_configuration()
+    {
+        chdir('tests/Fixtures/e2e/Unconfigured/');
+
+        $output = $this->runInfection(self::EXPECT_ERROR);
+
+        $this->assertContains(ConfigureCommand::NONINTERACTIVE_MODE_ERROR, $output);
+    }
+
+    /**
+     * @dataProvider e2eTestSuiteDataProvider
+     * @group e2e
+     */
+    public function test_it_runs_an_e2e_test_with_success(string $e2eTestName, string $fullPath)
+    {
+        $this->runOnE2EFixture($fullPath);
+    }
+
+    public function e2eTestSuiteDataProvider(): \Generator
+    {
+        $directories = Finder::create()
+            ->depth('== 0')
+            ->in(__DIR__ . '/../Fixtures/e2e/')
+            ->directories();
+
+        foreach ($directories as $dirName) {
+            if (file_exists($dirName . '/run_tests.bash')) {
+                // skipping non-standard tests
+                // specifically Memory_Limit - it is very slow to fail
+                continue;
+            }
+
+            yield [
+                // Makes PHPUnit report readable errors on failure:
+                // ...with data set #0 ('Foo_bar', ...
+                basename((string) $dirName),
+                $dirName,
+            ];
+        }
+    }
+
+    protected function tearDown()
+    {
+        if ($this->previousLoader) {
+            $this->previousLoader->unregister();
+        }
+
+        chdir($this->cwd);
+    }
+
+    private function runOnE2EFixture($path): string
+    {
+        $this->assertTrue(is_dir($path));
+        chdir($path);
+
+        $this->installComposerDeps();
+        $output = $this->runInfection(self::EXPECT_SUCCESS);
+
+        $this->assertRegExp('/You are running Infection with \w+ enabled./', $output);
+        $this->assertRegExp('/\d+ mutations were generated/', $output);
+        $this->assertRegExp('/\d+ mutants were killed/', $output);
```

With this a test it'll fail early if there's an error in the output.